docs/docs/getting_started/hello_noir/index.md

```diff
@@ -111,34 +111,41 @@ nargo execute witness-name
 
 The witness corresponding to this execution will then be written to the file `./target/witness-name.gz`.
 
-## Prove Our Noir Program
+The command also automatically compiles your Noir program if it was not already / was edited, which you may notice the compiled artifacts being written to the file `./target/hello_world.json`.
+
+## Proving Backend
 
-:::info
+Proving backends provide the ability to generate and verify proofs of executing Noir programs, following Noir's tooling that compiles and executes the programs. Read the [proving backend installation](../backend/index.md) section to learn more about proving backends and how to install them.
 
-Nargo no longer handles communicating with backends in order to generate proofs. In order to prove/verify your Noir programs, you'll need an installation of [bb](../barretenberg/index.md).
+Barretenberg is used as an example here to demonstrate how proving and verifying could be implemented and used. Read the [`bb` installation](../backend/index.md#example-installing-bb) section for how to install Barretenberg's CLI tool; refer to [`bb`'s documentations](https://github.com/AztecProtocol/barretenberg/blob/master/cpp/src/barretenberg/bb/readme.md) for full details about the tool.
 
-:::
+### 5. Prove an execution of the Noir program
 
-Prove the valid execution of your Noir program using `bb`:
+Using Barretenberg as an example, prove the valid execution of your Noir program running:
 
 ```sh
-bb prove -b ./target/hello_world.json -w ./target/witness-name.gz -o ./proof
+bb prove -b ./target/hello_world.json -w ./target/witness-name.gz -o ./target/proof
 ```
 
-A new file called `proof` will be generated in your project directory, containing the generated proof for your program.
+The proof generated will then be written to the file `./target/proof`.
 
-## Verify Our Noir Program
+### 6. Verify the execution proof
 
 Once a proof is generated, we can verify correct execution of our Noir program by verifying the proof file.
 
-Verify your proof by running:
+Using Barretenberg as an example, generate a verification key for your Noir program running:
```

Comment:
```suggestion
Using Barretenberg as an example, compute/output the verification key for the Noir program by running:
```
As I understand it, the proving key would already be generated, and the verification key is calculated from it.